### ISSUE TITLE
bootc image requirements

### ISSUE THREAD

**Issue by jmpolom** [containers/bootc#869]
bootc claims to be able to use an "OCI format" archive/image to deploy a system and lifecycle the thing through its paces. We all know that is glossing over some subtle but super important details concerning exactly how that image is structured and what content it has. This needs to be documented and as far as I can tell it really is not documented at all ([current docs](https://containers.github.io/bootc/building/guidance.html) do not do this subject justice).

**Comment by ryanabx** [comment]
This is probably the best page I could find, but it definitely could be improved on:

https://containers.github.io/bootc/bootc-images.html

**Comment by cgwalters** [comment]
Yes, it's in a far from ideal state but we're working on it. A few recent related PRs:

- https://github.com/containers/bootc/pull/888
- https://github.com/containers/bootc/pull/889

I also moved https://github.com/containers/bootc/pull/887 here which will dramatically reduce the requirements.

**Comment by cgwalters** [comment]
Jon most of what is happening in that kernel-related bits is related to the fact that there's some changes we want to make to the dracut invocation and we're basically just regenerating the initramfs and nuking the one from `/boot`. That bit of tech debt is on track to be significantly diminished with https://github.com/coreos/rpm-ostree/pull/5135 which will also make it much more obvious how to use `dnf upgrade kernel` etc.

**Comment by jmpolom** [comment]
> Jon most of what is happening in that kernel-related bits is related to the fact that there's some changes we want to make to the dracut invocation and we're basically just regenerating the initramfs and nuking the one from /boot.

I gathered that much. The issue for me is the intended outcome isn't documented so I'm left to just infer intention from source.

> That bit of tech debt is on track to be significantly diminished with https://github.com/coreos/rpm-ostree/pull/5135 which will also make it much more obvious how to use dnf upgrade kernel etc.

How will this impact bootc? Is bootc effectively going to be able to handle kernels that are in their normal places? It didn't to me seem like bootc has any special handling for kernels today.

**Comment by cgwalters** [comment]
> Is bootc effectively going to be able to handle kernels that are in their normal places? It didn't to me seem like bootc has any special handling for kernels today.

Does https://github.com/containers/bootc/pull/889 help on this? "normal" could mean different things to different people but that's the protocol bootc (and ostree) have expected for years.

**Comment by septatrix** [comment]
It would be fantastic if it were possible to generate a compatible image using mkosi which is used for building OS images and supports OCI archives as an output format
